qa/tasks/cephadm.py

```diff
@@ -1133,3 +1134,49 @@ def task(ctx, config):
 
         finally:
             log.info('Teardown begin')
+
+
+def add_mirror(s, mirror):
+    import toml
+    config = toml.loads(s)
+    is_v1 = 'registries' in config
+    if is_v1:
+        search = config.get('registries', {}).get('search', {}).get('registries', [])
+        insecure = config.get('registries', {}).get('search', {}).get('insecure', [])
+        v2 = {
+            'unqualified-search-registries': search,
+            'registry': [
+                {
+                    'prefix': reg,
+                    'location': reg,
+                    'insecure': reg in insecure,
+                    'blocked': False,
+                } for reg in search
+            ]
+        }
+    else:
+        v2 = config
+    dockers = [r for r in v2['registry'] if r['prefix'] == 'docker.io']
+    if dockers:
+        docker = dockers[0]
+        docker['mirror'] = [{
+            "location": "vossi04.front.sepia.ceph.com:5000",
+            "insecure": True,
+        }]
+    return v2
+
+def add_mirror_to_cluster(ctx, mirror):
+    # add other hosts
+    for remote in ctx.cluster.remotes.keys():
+        log.info('Add caching registry')
+        config = teuthology.get_file(
+            remote=remote,
+            path='/etc/containers/registry.conf'
+        )
+        new_config = toml.dumps(add_mirror(config))
```

Comment:
This overall code should go to teuthology repo as something like `teuthology.util.podman`